src/VirtualClient/VirtualClient.Actions/OpenFOAM/OpenFOAMExecutor.cs

```diff
@@ -302,13 +302,13 @@ await process.StartAndWaitAsync(cancellationToken)
 
                 if (!cancellationToken.IsCancellationRequested)
                 {
-                    if (process.IsErrored())
+                    await this.LogProcessDetailsAsync(process, telemetryContext, "OpenFOAM", logToFile: true);
+                    process.ThrowIfWorkloadFailed();
+
+                    if (!arguments.Contains("clean"))
```

Comment:
Please add a comment explaining why you are choosing a word "clean" instead of other indicators like exit code.

Reply:
added

Reply:
Did you push that? or you added else where?

Reply:
it is in the latest commit, i think u checked last commit.